examples/sst2/train.py

```diff
@@ -0,0 +1,352 @@
+    label: The correct labels.
+
+  Returns:
+    The binary cross entropy loss for each given logit.
+  """
+  return label * nn.softplus(-logit) + (1 - label) * nn.softplus(logit)
+
+
+@jax.jit
+def train_step(optimizer: Any, inputs: jnp.ndarray, lengths: jnp.ndarray,
+               labels: jnp.ndarray, rng: Any, l2_reg: float):
+  """Single optimized training step.
+
+  Args:
+    optimizer: The optimizer to use to update the weights.
+    inputs: A batch of inputs. <int64>[batch_size, seq_len]
+    lengths: The lengths of the sequences in the batch. <int64>[batch_size]
+    labels: The labels of the sequences in the batch. <int64>[batch_size, 1]
+    rng: Random number generator for dropout.
+    l2_reg: L2 regularization weight.
+
+  Returns:
+    optimizer: The optimizer in its new state.
+    loss: The loss for this step.
+  """
+  rng, new_rng = jax.random.split(rng)
+  def loss_fn(model):
+    with nn.stochastic(rng):
+      logits = model(inputs, lengths, train=True)
+    loss = jnp.mean(binary_cross_entropy_loss(logits, labels))
+
+    # L2 regularization
+    l2_params = jax.tree_leaves(model.params['lstm_classifier'])
+    l2_weight = jnp.sum([jnp.sum(p ** 2) for p in l2_params])
+    l2_penalty = l2_reg * l2_weight
+
+    loss = loss + l2_penalty
+    return loss, logits
+
+  loss, _, grad = optimizer.compute_gradient(loss_fn)
+  optimizer = optimizer.apply_gradient(grad)
+  return optimizer, loss, new_rng
+
+
+def get_predictions(logits: jnp.ndarray) -> jnp.ndarray:
+  """Returns predictions given a batch of logits."""
+  outputs = jax.nn.sigmoid(logits)
+  return (outputs > 0.5).astype(jnp.int32)
+
+
+def get_num_correct(logits: jnp.ndarray, labels: jnp.ndarray) -> jnp.ndarray:
+  """Returns the number of correct predictions."""
+  return jnp.sum(get_predictions(logits) == labels)
+
+
+@jax.jit
+def eval_step(model: nn.Module, inputs: jnp.ndarray, lengths: jnp.ndarray,
+              labels: jnp.ndarray):
+  """A single evaluation step.
+
+  Args:
+    model: The model to be used for this evaluation step.
+    inputs: A batch of inputs. <int64>[batch_size, seq_len]
+    lengths: The lengths of the sequences in the batch. <int64>[batch_size]
+    labels: The labels of the sequences in the batch. <int64>[batch_size, 1]
+
+  Returns:
+    loss: The summed loss on this batch.
+    num_correct: The number of correct predictions in this batch.
+  """
+  logits = model(inputs, lengths, train=False)
+  loss = jnp.sum(binary_cross_entropy_loss(logits, labels))
+  num_correct = get_num_correct(logits, labels)
+  return loss, num_correct
+
+
+def evaluate(model: nn.Model, eval_ds: tf.data.Dataset):
+  """Evaluates the model on a dataset.
+
+  Args:
+    model: A model to be evaluated.
+    eval_ds: A dataset to be used for the evaluation. Typically valid or test.
+
+  Returns:
+    A dict with the evaluation results.
+  """
+  count = 0
+  total_loss = 0.
+  total_correct = 0
+
+  for ex in tfds.as_numpy(eval_ds):
```

Comment:
Would not pass google python review :) No abbreviations that are not standard.

Reply:
Fixed.